Svc/FileDownlink/FileDownlink.cpp

```diff
@@ -429,7 +430,12 @@ namespace Svc {
     filePacket.fromCancelPacket(cancelPacket);
     this->getBuffer(buffer, CANCEL_PACKET);
 
-    const Fw::SerializeStatus status = filePacket.toBuffer(buffer);
+    // Serialize the packet descriptor FW_PACKET_FILE to the buffer
+    Fw::SerializeStatus status = buffer.getSerializer().serialize(Fw::ComPacket::FW_PACKET_FILE);
+    FW_ASSERT(status == Fw::FW_SERIALIZE_OK);
+    Fw::Buffer offsetBuffer(buffer.getData() + sizeof(FwPacketDescriptorType), filePacket.bufferSize());
```

Comment:
You are setting the buffer size here to `filePacket.bufferSize();` (just like in `sendFilePacket`), however; unlike in `sendFilePacket` there is no assert to ensure this size is correct.

You should either:
1.  Set the size of `offsetBuffer` to be `buffer.getSize() - sizeof(FwPacketDescriptor()` --or--
2. Add an assert like in `sendFilePacket`
3. Do both

I think 3 is probably the safest.

Reply:
Done!